lib/async_hooks.js

```diff
@@ -286,15 +286,15 @@ class AsyncResource {
 
     this[async_id_symbol] = ++async_id_fields[kAsyncIdCounter];
     this[trigger_async_id_symbol] = triggerAsyncId;
-    // this prop name (destroyed) has to be synchronized with async-wrap.cc
+    // this prop name (destroyed) has to be synchronized with the string further down
     this[destroyedSymbol] = { destroyed: false };
 
     emitInitScript(
       this[async_id_symbol], type, this[trigger_async_id_symbol], this
     );
 
     if (!opts.requireManualDestroy) {
-      registerDestroyHook(this, this[async_id_symbol], this[destroyedSymbol]);
+      registerDestroyHook(this, this[async_id_symbol], this[destroyedSymbol], "destroyed");
```

Comment:
Typically we solve the performance penalty of creating a string in C++ by having it as a global. We have a lot of those so don't feel bad about adding another one: https://github.com/nodejs/node/blob/master/src/env.h#L100